addons/test_mail/tests/test_mail_activity.py

```diff
@@ -215,6 +231,150 @@ def test_action_feedback_attachment(self):
             self.assertEqual(attachment.res_id, activity_message.id)
             self.assertEqual(attachment.res_model, activity_message._name)
 
+    def test_mail_activity_state(self):
```

Comment:
Could be great to add a docstring in tests when dealing with timezone. A small graph of: user, timezones, expected datetime / dates, ... otherwise you end up with people just changing the result of your test when it is red because they don't understand it anymore :) .